src/extensionApi.ts

```diff
@@ -79,6 +81,41 @@ export class CommandHandler {
         }
     }
 
+    public async debugServer(mode: string, context?: Protocol.ServerState): Promise<Protocol.StartServerResponse> {
+
+        const folderProject = await vscode.window.showOpenDialog({
+            canSelectFiles: true,
+            canSelectMany: false,
+            canSelectFolders: true,
+            openLabel: 'Open Project to debug'
+        } as vscode.OpenDialogOptions).then(
+            folder => {
+                if (folder != null && folder.length > 0) {
+                    vscode.commands.executeCommand('vscode.openFolder', folder[0]);
+                    return folder[0];
+                }
+            }
+        );
+
+        if (!folderProject) return;
+
+        if (vscode.extensions.getExtension('vscjava.vscode-java-debug') === undefined) {
+            vscode.window.showErrorMessage('Debugger for Java extension is required. Install/Enable it before proceeding.');
+            return;
+        }
+
+        return this.startServer(mode, context).then(
+            status => {
+                const debugDetails = status.details.properties;
+                const workspaceFolder = vscode.workspace.getWorkspaceFolder(folderProject);
+                vscode.debug.startDebugging(workspaceFolder, new CustomDebugConfiguration(debugDetails['debug.details.port']).provideDebugConfigurations(workspaceFolder)[0]);
```

Comment:
Why do we need to implement a specific interface if we call the method explicitely ?

Reply:
In this way we can manage the debugging session in a better way. During the test i made, it happened that the server could be started correctly but when i attached the debugger it stopped for some reason. It resulted in a situation where the debug seemed to be started and the server was stopped (i will open an issue on rsp-server soon). By implementing that specific interface, I read, we can work with cancellation tokens and stop the debugging process programmatically. I have not implemented it yet but I think can be useful for such situations.

BTW you're correct, we could just pass the custom debug configuration to the method without implementing the interface to make it work.